src/main/java/com/icthh/xm/ms/entity/service/LocationService.java

```diff
@@ -53,7 +63,48 @@
      */
     public Location save(Location location) {
         log.debug("Request to save location : {}", location);
-        location.setXmEntity(xmEntityRepository.getOne(location.getXmEntity().getId()));
+
+        Objects.nonNull(location);
+        Objects.nonNull(location.getXmEntity());
+        Objects.nonNull(location.getXmEntity().getId());
+
+        XmEntity entity = xmEntityRepository.findById(location.getXmEntity().getId()).orElseThrow(
+            () -> new EntityNotFoundException("No entity found by id: " + location.getXmEntity().getId())
+        );
+
+        LocationSpec spec = getSpec(entity, location);
+
+        if (location.getId() == null && spec.getMax() != null) {
+            //forbid to add element if spec.max = 0
+            assertZeroRestriction(spec);
+            //forbid to add element if spec.max <= addedSize
+            assertLimitRestriction(spec, entity);
+        }
+
+        location.setXmEntity(entity);
         return locationRepository.save(location);
     }
+
+    protected LocationSpec getSpec(XmEntity entity, Location location) {
+        Objects.nonNull(entity);
```

Comment:
`Objects.nonNull` is just check for null.
Use `Objects.requireNonNul(T)` in case you need to throw exception.

Reply:
fixed